### DISCUSSION TITLE
AWS Athena support

### DISCUSSION THREAD

**Discussion by rabidaudio** [evidence-dev/evidence#2352] [Data Source Request] [answered]
**What do you use this database for?**
I have a small hobbyist data project, and for that Athena is incredibly cost-effective.

**Are there any popular and well-maintained clients for this database?**
The official connector is `@aws-sdk/client-athena`, there's also `athena-express` which hides some of the [clunkiness of the official one](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-athena/globals.html).

**Are you interested in contributing this database connector?**
Draft PR incoming

**Comment by archiewood** [comment]
Hello -looping back on connector requests

We have recently re-architected Evidence (since this issue was opened), and Evidence is now "pluggable", meaning you can write your own connectors, publish them, and use them in Evidence.

If you are interesting in writing a connector, here are some resources:

1. Docs on creating a plugin: https://docs.evidence.dev/plugins/creating-a-plugin/datasources
2. Template for a connector plugin: https://github.com/evidence-dev/datasource-template
3. An example that I spun up for pulling stats from the npm API: https://github.com/archiewood/npm-stats

If you do write a plugin:

- the suggested package name is `evidence-connector-[plugin]`
- the docs are new, so let us know if there is extra information you need
- share it on the Evidence [community](https://github.com/evidence-dev/evidence/issues/slack.evidence.dev), and we'll feature it!

**Comment by hughess** [comment, marked as answer]
@morganchorlton3 has just published an Athena plugin here:

Repo: https://github.com/morganchorlton3/evidence-connector-aws-athena
NPM: https://www.npmjs.com/package/evidence-connector-aws-athena
